frontend/src/components/widgets/CameraInput/WebcamComponent.tsx

```diff
@@ -0,0 +1,174 @@
+    WebcamPermission.PENDING
+  )
+  const videoRef = useRef<Webcam>(null)
+  const [facingMode, setFacingMode] = useState(FacingMode.USER)
+
+  function capture(): void {
+    if (videoRef.current !== null) {
+      const imageSrc = videoRef.current.getScreenshot()
+      handleCapture(imageSrc)
+    }
+  }
+
+  function switchCamera(): void {
+    setFacingMode(prevState =>
+      prevState === FacingMode.USER ? FacingMode.ENVIRONMENT : FacingMode.USER
+    )
+  }
+
+  const theme: Theme = useTheme()
+
+  return (
+    <StyledCameraInput width={width}>
+      {webcamPermission !== WebcamPermission.SUCCESS &&
+      !disabled &&
+      !inClearPhotoMode ? (
+        <AskForCameraPermission width={width} />
+      ) : (
+        isMobile && (
+          <StyledSwitchFacingModeButton>
+            <Tooltip content={"Switch camera"} placement={Placement.TOP_RIGHT}>
+              <Button kind={Kind.MINIMAL} onClick={switchCamera}>
+                <Icon
+                  content={SwitchCamera}
+                  size="twoXL"
+                  color={themeColors.white}
+                />
+              </Button>
+            </Tooltip>
+          </StyledSwitchFacingModeButton>
+        )
+      )}
+      <StyledBox
+        hidden={
+          webcamPermission !== WebcamPermission.SUCCESS &&
+          !disabled &&
+          !inClearPhotoMode
+        }
+        width={width}
+      >
+        {!disabled && (
+          <Webcam
+            audio={false}
+            ref={videoRef}
+            screenshotFormat="image/jpeg"
+            screenshotQuality={1}
+            width={width}
+            style={{
+              borderRadius: `${theme.radii.md} ${theme.radii.md} 0 0`,
+            }}
+            height={(width * 9) / 16}
+            onUserMediaError={() => {
+              setWebcamRequestState(WebcamPermission.ERROR)
+            }}
+            onUserMedia={() => {
+              setWebcamRequestState(WebcamPermission.SUCCESS)
+              setInClearPhotoMode(false)
+            }}
+            videoConstraints={{
+              width: { ideal: width },
+              facingMode,
+            }}
+          />
+        )}
+      </StyledBox>
+      <CameraInputButton
+        onClick={capture}
+        disabled={
+          webcamPermission !== WebcamPermission.SUCCESS ||
+          disabled ||
+          inClearPhotoMode
+        }
+      >
+        Take Photo
+      </CameraInputButton>
+    </StyledCameraInput>
+  )
+}
+
+interface AskForCameraPermissionProps {
+  width: number
+}
+
+function AskForCameraPermission({
+  width,
+}: AskForCameraPermissionProps): ReactElement {
+  return (
+    <StyledBox width={width}>
+      <Icon size="threeXL" color={themeColors.gray60} content={Video} />
+      <StyledDescription>
+        This app would like to use your camera.
+        <StyledLink href="https://streamlit.io">
+          Learn how to allow access.
+        </StyledLink>
```

Comment:
I'm assuming this link is a placeholder?

Also, I think product would probably want to make a pass on the copy for this